packages/@tailwindcss-upgrade/src/codemods/migrate-tailwind-directives.test.ts

```diff
@@ -0,0 +1,172 @@
+import dedent from 'dedent'
+import postcss from 'postcss'
+import { expect, it } from 'vitest'
+import { migrateTailwindDirectives } from './migrate-tailwind-directives'
+
+const css = dedent
+
+function migrate(input: string) {
+  return postcss()
+    .use(migrateTailwindDirectives())
+    .process(input, { from: expect.getState().testPath })
+    .then((result) => result.css)
+}
+
+it("should not migrate `@import 'tailwindcss'`", async () => {
+  expect(
+    await migrate(css`
+      @import 'tailwindcss';
+    `),
+  ).toEqual(css`
+    @import 'tailwindcss';
+  `)
+})
+
+it('should migrate the default @tailwind directives to a single import', async () => {
+  expect(
+    await migrate(css`
+      @tailwind base;
+      @tailwind components;
+      @tailwind utilities;
+    `),
+  ).toEqual(css`
+    @import 'tailwindcss';
+  `)
+})
+
+it('should migrate the default @tailwind directives as imports to a single import', async () => {
+  expect(
+    await migrate(css`
+      @import 'tailwindcss/base';
+      @import 'tailwindcss/components';
+      @import 'tailwindcss/utilities';
+    `),
+  ).toEqual(css`
+    @import 'tailwindcss';
+  `)
+})
+
+it.each([
+  [
+    css`
+      @tailwind components;
+      @tailwind base;
+      @tailwind utilities;
+    `,
+  ],
+  [
+    css`
+      @tailwind components;
+      @tailwind utilities;
+      @tailwind base;
+    `,
+  ],
+  [
+    css`
+      @tailwind utilities;
+      @tailwind base;
+      @tailwind components;
+    `,
+  ],
+  [
+    css`
+      @tailwind utilities;
+      @tailwind components;
+      @tailwind base;
```

Comment:
I'm not sure that this translation is correct — though maybe it's still fine?

In this case in v3 utilities would definitely appear _before_ base layer styles (though I can't think of any reason anyone would ever do that). `@import 'tailwindcss'` would definitely not operate that way.

Reply:
The order shouldn't matter anymore since we are using native layers and `@import 'tailwindcss'` defines the order at the top:
```css
/* tailwindcss/index.css */
@layer theme, base, components, utilities;

@import './theme.css' layer(theme);
@import './preflight.css' layer(base);
@import './utilities.css' layer(utilities);
```

I think if we want to be 100% correct here, we should a modified version of `@layer theme, base, components, utilities;` at the top.

Thoughts?

Reply:
If we have multiple `@layer foo, bar, baz` definitions, then the first one wins. E.g.:

https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Atrue%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22%40layer%20base%2C%20theme%2C%20components%2C%20utilities%3B%5Cn%40layer%20theme%2C%20components%2C%20base%2C%20utilities%3B%5Cn%40layer%20theme%2C%20base%2C%20components%2C%20utilities%3B%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D

Reply:
I think that would be the most correct translation but uh maybe it's fine. Maybe @adamwathan has thoughts here? It could help us nudge people to order them in a more consistent way.

Reply:
Added support for that in the latest commit: https://github.com/tailwindlabs/tailwindcss/pull/14411/commits/5796dadb9957d116caa19890e077f700c7fe98ca